tests/test_traindiffs.py

```diff
@@ -0,0 +1,91 @@
+"""
+Runs 10 steps of SGD for each workload and compares results.
+Run it as:
+  python3 test_traindiffs.py
+"""
+import pickle
+from subprocess import DEVNULL
+from subprocess import run
+from subprocess import STDOUT
+
+from absl import flags
+from absl.testing import absltest
+import jax
+
+jax.config.update('jax_platforms', 'cpu')
+
+FLAGS = flags.FLAGS
+
+WORKLOADS = [
+    'imagenet_resnet',
+    'imagenet_vit',
+    'wmt',
+    'librispeech_conformer',
+    'librispeech_deepspeech',
+    'fastmri',
+    'ogbg',
+]
+GLOBAL_BATCH_SIZE = 16
+
+
+class ReferenceSubmissionTest(absltest.TestCase):
+
+  def test_workload(self):
+    # pylint: disable=line-too-long, unnecessary-lambda-assignment
+    for workload in WORKLOADS:
+      name = f'Testing {workload}'
+      jax_logs = '/tmp/jax_log.pkl'
+      pyt_logs = '/tmp/pyt_log.pkl'
+      run(f'XLA_PYTHON_CLIENT_ALLOCATOR=platform python3 tests/modeldiffs/train_diff.py --workload={workload} --framework=jax --global_batch_size={GLOBAL_BATCH_SIZE} --log_file={jax_logs}',
```

Comment:
instead of having to go through the shell and back, I think it would be cleaner if we rewrote test_traindiffs.py (or reference_algorithm_tests.py if we merge them) to have a function that takes the inputs and returns the outputs we need here. assuming we want to keep the torch dpp setup for the test, really we just need a way to run the test in torchrun. maybe we can call this directly https://github.com/pytorch/pytorch/blob/master/torch/distributed/run.py ? unfortunately I don't think that returns anything though. really wish it was cleaner to call pytorch on a multi-device setup.

for now, can you add a TODO here describing this issue?

Reply:
I added some info into the documentation-string of this test method, can you please confirm if that is okay?